--- a/Quake/sv_phys.c
+++ b/Quake/sv_phys.c
@@ -1170,6 +1170,7 @@
 {
 	int	i;
 	int	entity_cap; // For sv_freezenonclients 
+	int	*pentitycap;
 	edict_t	*ent;
 
 // let the progs know that a new frame has started
@@ -1186,12 +1187,14 @@
 	ent = sv.edicts;
 
 	if (sv_freezenonclients.value)
+	{
 	  entity_cap = svs.maxclients + 1; // Only run physics on clients and the world
+	  pentitycap = &entity_cap;
+	}
 	else
-	  entity_cap = sv.num_edicts;
+	  pentitycap = &sv.num_edicts;
 
-	//for (i=0 ; i<sv.num_edicts ; i++, ent = NEXT_EDICT(ent))
-	for (i=0 ; i<entity_cap ; i++, ent = NEXT_EDICT(ent))
+	for (i = 0; i < *pentitycap; ++i, ent = NEXT_EDICT(ent))
 	{
 		if (ent->free)
 			continue;
