--- a/Quake/net_bsd.c
+++ b/Quake/net_bsd.c
@@ -66,12 +66,14 @@
 const int net_numdrivers = Q_COUNTOF(net_drivers);
 
 #include "net_udp.h"
+#include "net_qn.h"
 
 net_landriver_t	net_landrivers[] =
 {
 	{	"UDP",
 		false,
 		0,
+		0,
 		UDP_Init,
 		UDP_Shutdown,
 		UDP_Listen,
@@ -90,6 +92,30 @@
 		UDP_AddrCompare,
 		UDP_GetSocketPort,
 		UDP_SetSocketPort
+	},
+
+	{	"QN",
+		false,
+		0,
+		QN_DATAGRAM_MAX,
+		QN_Init,
+		QN_Shutdown,
+		QN_Listen,
+		QN_OpenSocket,
+		QN_CloseSocket,
+		QN_Connect,
+		QN_CheckNewConnections,
+		QN_Read,
+		QN_Write,
+		QN_Broadcast,
+		QN_AddrToString,
+		QN_StringToAddr,
+		QN_GetSocketAddr,
+		QN_GetNameFromAddr,
+		QN_GetAddrFromName,
+		QN_AddrCompare,
+		QN_GetSocketPort,
+		QN_SetSocketPort
 	}
 };
 
