--- a/Quake/net_main.c
+++ b/Quake/net_main.c
@@ -60,7 +60,7 @@
 int		unreliableMessagesReceived	= 0;
 
 static	cvar_t	net_messagetimeout = {"net_messagetimeout","300",CVAR_NONE};
-cvar_t	hostname = {"hostname", "UNNAMED", CVAR_NONE};
+cvar_t	hostname = {"hostname", "UNNAMED", CVAR_ARCHIVE};
 
 // these two macros are to make the code more readable
 #define sfunc	net_drivers[sock->driver]
--- a/Quake/net_loop.c
+++ b/Quake/net_loop.c
@@ -55,10 +55,10 @@
 
 	hostCacheCount = 1;
 	if (Q_strcmp(hostname.string, "UNNAMED") == 0)
-		Q_strcpy(hostcache[0].name, "local");
+		q_strlcpy(hostcache[0].name, "local", sizeof(hostcache[0].name));
 	else
-		Q_strcpy(hostcache[0].name, hostname.string);
-	Q_strcpy(hostcache[0].map, sv.name);
+		q_strlcpy(hostcache[0].name, hostname.string, sizeof(hostcache[0].name));
+	q_strlcpy(hostcache[0].map, sv.name, sizeof(hostcache[0].map));
 	hostcache[0].users = net_activeconnections;
 	hostcache[0].maxusers = svs.maxclients;
 	hostcache[0].driver = net_driverlevel;
